libmozevent/storage.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+import math
+import os
+import pickle
+import time
+
+import structlog
+
+from libmozevent.utils import AsyncRedis
+
+logger = structlog.get_logger(__name__)
+
+
+class EphemeralStorage:
+    @classmethod
+    async def create(cls, name, expiration):
+        self = EphemeralStorage()
+        self.name = name
+        self.expiration = expiration
+        self.cache = {}
+        self.redis_enabled = "REDIS_URL" in os.environ
+        logger.info("Redis support", enabled=self.redis_enabled and "yes" or "no")
+
+        if self.redis_enabled:
+            async with AsyncRedis() as redis:
+                # Remove all expired keys from our sorted set.
+                await redis.zremrangebyscore(
+                    self.name, min=-math.inf, max=int(time.time())
+                )
+                # Load all live keys from our sorted set.
+                keys = await redis.zrangebyscore(
+                    self.name, min=int(time.time()), max=math.inf
+                )
+                for key in keys:
+                    key = key.decode("ascii")
```

Comment:
Why ascii ?

Reply:
I was expecting the keys to be simpler, but we can use utf-8 to be safe.